### DISCUSSION TITLE
Make file browser more intuitive

### DISCUSSION THREAD

**Discussion by gornostal** [Ulauncher/Ulauncher#892] [Ideas]
It's not obvious that user has to:
* start query with `~` or `/` to see the file browser
* press `Alt+Enter` to open an alt menu of a file/folder

**Comment by geckolinux** [comment]
I think that files and folders mixed in with application results is the best option, especially when combined with good adaptive intelligence based on frequency of use. For example, in my case, I would expect `cl` to bring up in this order:

1. ~/Documents/**cl**ients.odt
2. **Cl**ementine music player
3. Thunderbird mail **cl**ient

**Reply by githorse** [reply]
+1 for 1) file results mixed in with application results, and 2) fuzzy matching on files (without requiring an extension). Coming from Alfred I'm definitely missing this.

**Comment by gornostal** [comment]
That would be an ideal solution.
Not so easy to implement however.

I'll take this suggestion into account when will be creating plugin for searching files.
In this ticket I only wanted to make file browsing more intuitive.

**Comment by geckolinux** [comment]
Maybe you could borrow some ideas from [Albert](https://github.com/albertlauncher/albert), which does this quite well in my opinion. (It's method of updating the index isn't ideal, however.)

**Comment by orschiro** [comment]
@geckolinux maybe Albert and uLauncher — both are fantastic launchers — should just merge into one even greater project. :-)

**Comment by ibrokemypie** [comment]
In my opinion, the most useful would be to mix application and file searches, but allowing users to rank which search engine is most important, so if they search for something, application results show up on top with file results below them for example.

**Comment by simonwiles** [comment]
...and if extensions could add their results in with applications and file searches, that would be really great :)

**Comment by grahamjpark** [comment]
On a somewhat related note, I think having "Open Folder" and "Copy Path to Folder" being the top two options while typing would help a lot. Took me a long time to figure out how to open up a folder from the file search.

**Comment by amontalenti** [comment]
Agree with this. I also don't really understand why pressing `~` auto-populates as `~/`, or why simply typing `~/some_folder <ENTER>` (e.g. `~/Documents <Enter>`) doesn't just open that directory in your file manager.
